org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilter.java

```diff
@@ -94,7 +95,16 @@ private void match(final MethodNode methodNode,
 						"getCOROUTINE_SUSPENDED", "()Ljava/lang/Object;");
 			}
 
-			nextIsVar(Opcodes.ASTORE, "COROUTINE_SUSPENDED");
+			if (cursor != null
+					&& Opcodes.POP == skipNonOpcodes(cursor.getNext())
+							.getOpcode()) {
+				nextIs(Opcodes.POP);
+				vars.put("COROUTINE_SUSPENDED",
```

Comment:
This is not fully clear from the PR why this line is needed.

After discussing it, I suggest either adding a comment that this is a safe guard or refactoring the code, so you don't need it.

Reply:
@leveretka ok, I did refactoring